### ISSUE TITLE
Checkbox - Accessibility Issue in brn-checkbox Component

### ISSUE THREAD

**Issue by tavivibrazowski** [spartan-ng/spartan#644]
### Please provide the environment you discovered this bug in.

"@angular/core": "18.2.10"
"@spartan-ng/brain": "0.0.1-alpha.404"

### Which area/package is the issue in?

checkbox

### Description

The brn-checkbox component is failing accessibility tests due to improper implementation of ARIA attributes.

**Issue Description**
When running accessibility tests on the checkbox component, the following error is reported:
`'aria-conditional-attr' | 'serious' | "Ensure ARIA attributes are used as described in the specification of the element's role"`

**Root Cause**
The input element in the checkbox component has conflicting role and state management attributes:

1. Native <input type="checkbox"> is being used
2. Redundant explicit role="checkbox" is added
3. Using aria-checked="true" instead of the native checked attribute.

`<input tabindex="-1" type="checkbox" role="checkbox" value="on" id="default-checkbox" name="default-checkbox" aria-label="Default Checkbox" aria-checked="true" style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;">`

**Expected Behavior**
The checkbox should comply with ARIA specifications, which state that when using native HTML elements, their implicit roles should be relied upon rather than explicitly declared, and their native state attributes should be used instead of ARIA equivalents when available.

### Please provide the exception or error you saw

```true
`'aria-conditional-attr' | 'serious' | "Ensure ARIA attributes are used as described in the specification of the element's role"`
```

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [x] No

**Comment by goetzrobin** [comment]
Hey @tavivibrazowski I want to make sure we are getting this right. Are you suggesting to remove the role="checkbox" attribute and add/remove the checked attribute in addition to the aria-checked attribute? Thanks already!

**Comment by tavivibrazowski** [comment]
The `role=checkbox` is redundant as you already defined the input with `type=checkbox`.
In addition, your should use the [checked attribute](http://www.w3schools.com/TAGs/att_input_checked.asp) for both testability and accessibility.
